### PR TITLE
Approve applications in transaction

### PR DESCRIPTION
**Pull request by Nothing4You**
Use transaction when approving applications to ensure consistent approval state

depends on #4923, required for #4907

### CODE REVIEW

File: crates/api/src/site/registration_applications/approve.rs [hunk 1]
```diff
@@ -25,41 +29,66 @@ pub async fn approve_registration_application(
...
-    .ok_or(LemmyErrorType::CouldntFindRegistrationApplication)?;
+        Ok::<_, LemmyError>((
+          approval_email_user_view,
+          Json(RegistrationApplicationResponse {
+            registration_application,
+          }),
+        ))
+      }) as _
+    })
+    .await?;
```
Comment (dullbananas) [inline crates/api/src/site/registration_applications/approve.rs:85]: Moving the things other than updates into the transaction doesn't remove the possibility of inconsistency, and I don't know if that's necessary.

**Review by dullbananas — CHANGES_REQUESTED** [review]

File: crates/api/src/site/registration_applications/approve.rs (hunk 1 above)
Comment (Nothing4You) [reply crates/api/src/site/registration_applications/approve.rs:85]: What do you mean?

The transaction spans from the first SQL query to the last one, so if anything in the middle fails the transaction gets rolled back.

File: crates/api/src/site/registration_applications/approve.rs [hunk 2]
```diff
@@ -25,41 +29,66 @@ pub async fn approve_registration_application(
   // Only let admins do this
   is_admin(&local_user_view)?;
 
-  // Update the registration with reason, admin_id
-  let deny_reason = diesel_string_update(data.deny_reason.as_deref());
-  let app_form = RegistrationApplicationUpdateForm {
-    admin_id: Some(Some(local_user_view.person.id)),
-    deny_reason,
-  };
+  let pool_context = context.clone();
```
[the remark this answers is not in the document]
Comment (Nothing4You) [reply crates/api/src/site/registration_applications/approve.rs]: leftover from earlier attempts, removed now.

File: crates/api/src/site/registration_applications/unread_count.rs [hunk 3]
```diff
@@ -1,4 +1,5 @@
-use actix_web::web::{Data, Json};
+use activitypub_federation::config::Data;
+use actix_web::web::Json;
```
[the remark this answers is not in the document]
Comment (Nothing4You) [reply crates/api/src/site/registration_applications/unread_count.rs:2]: those changes are from #4923, will rebase this once the other PR is merged

File: crates/api/src/site/registration_applications/approve.rs (hunk 1 above)
Comment (Nothing4You) [reply crates/api/src/site/registration_applications/approve.rs:84]: got it

Comment (Nothing4You) [reply crates/api/src/site/registration_applications/approve.rs:61]: should be good now

**Review by dullbananas — APPROVED** [review]

**Review by dessalines — APPROVED** [review]

**Pull request merged**